mat64/dense.go

```diff
@@ -164,13 +175,53 @@ func (m *Dense) View(i, j, r, c int) Matrix {
 	t.mat.Data = t.mat.Data[i*t.mat.Stride+j : (i+r-1)*t.mat.Stride+(j+c)]
 	t.mat.Rows = r
 	t.mat.Cols = c
+	t.capRows -= i
+	t.capCols -= j
 	return &t
 }
 
+func (m *Dense) Grow(r, c int) {
```

Comment:
I'm not sure what the cases are where you don't assign it to itself, but there are very few functions that self-modify in this package, so it seemed more consistent to return.